### PR TITLE
Add SAR swath information

### LINKED ISSUE #17: SAR beams/swath
This is particularly relevant for Stripmap where we want to know the beam and for TOPSAR, swaths.

For instance, a Sentinel-1 StripMap will include `<s1sarl1:swath>S1</s1sarl1:swath>` in the manifest and we get a value in [S1, S2, S3, S4, S4, S6]

For CSK Stripmap HIMAGE it's a value in H4-n where n is 0A, 0B, 1 to 24.

We believe it's a relevant use case both as a queryable and as part of the STAC Item sar extension properties.

Proposal: `sar:beam`

For S1 TOPSAR, the value could be [IW1, IW2, IW3]

Proposal: `sar:swaths`

### PR DESCRIPTION
**Pull request by emmanuelmathot**
This PR adds the field `sar:swath` composed of beam objects to describe precisely subswaths.

It fixes #17

### CODE REVIEW

**Comment by m-mohr** [comment]
@emmanuelmathot Is any of this something users would search on?

**Comment by fabricebrito** [comment]
@m-mohr yes, user searchs for S1 Stripmap with beam S6 to create an interferometric pair

**Comment by fabricebrito** [comment]
@emmanuelmathot I'd add an example of a Stripmap product in the README and stripmap-item.json. It eases the adoption

File: README.md [hunk 1]
```diff
@@ -61,36 +64,72 @@ for instance for `HH+HV` add both `HH` and `HV`.
...
+
+The beam object is used to describe the swath of the SAR acquisition.
+According to the instrument mode defined in `sar:instrument_mode`, the `sar:swath` is composed of one or more beams.
+
+| Field Name     | Type                  | Description                                                                            |
+| -------------- | --------------------- | -------------------------------------------------------------------------------------- |
+| id             | string                | **REQUIRED.** The unique identifier of the beam. It can be the subswath identifier     |
+| relative_burst | number                | Identification number that uniquely identifies a burst cycle within each repeat cycle. |
+| polarizations  | \[string]             | Any combination of polarizations.                                                      |
+| view:*         | View extension fields | Additional fields from the [View extension](https://github.com/stac-extensions/view)   |
```
Comment (m-mohr) [inline README.md:77]: I think in other extensions we just added text for such cases. The Type is also not quite a type... ;-)

```suggestion

It is recommended to enrich the Beam Object fields with fields from the [View extension](https://github.com/stac-extensions/view), e.g. the incidence angle.
```

File: README.md [hunk 2]
```diff
@@ -61,36 +64,72 @@ for instance for `HH+HV` add both `HH` and `HV`.
...
+
+> \[!NOTE]
+> If the value of a field repeats for all beams, it is recommended to include it only in the first beam.
```
Comment (m-mohr) [inline README.md:80]: This is weird to me. I assume the id and relative_burst would never be the same for all beams?
And the view fields and the polarizations should then just be on the upper level, e.g. in the Item properties.

```suggestion
> If the value of a field repeats for all beams, it is recommended to include it only in the parent scope, i.e. usually in the Item Properties.
```

File: README.md [hunk 3]
```diff
@@ -61,36 +64,72 @@ for instance for `HH+HV` add both `HH` and `HV`.
...
+  },
+  {
+    "id": "IW2",
+    "view:incident_angle": 34.0
+  },
+  {
+    "id": "IW3",
+    "view:incident_angle": 36.0
+  }
+]
```
Comment (m-mohr) [inline README.md:100]:
```suggestion
"sar:polarizations": ["VV", "VH"],
"sar:swath": [
  {
    "id": "IW1",
    "relative_burst": 541,
    "view:incident_angle": 32.0
  },
  {
    "id": "IW2",
    "view:incident_angle": 34.0
  },
  {
    "id": "IW3",
    "view:incident_angle": 36.0
  }
]
```

Maybe I'm misunderstanding something here, but wouldn't that make more sense?

**Comment by m-mohr** [comment]
> @m-mohr yes, user searchs for S1 Stripmap with beam S6 to create an interferometric pair

Okay. For search the currentl structure of an array of objects is not ideal. If we want to enable easy searching we should probably make the structure simpler or so.

**Comment by fabricebrito** [comment]
Yes, I agree with going for a simpler structure and keep the focus on discovery rather than extensive metadata in the STAC Item. 
Additional information can be found in the metadata.xml (or equivalent) of the SAR product.

**Comment by m-mohr** [comment]
Maybe just `sar:beam_ids: ["IW1", "IW2"]` (or `sar:swatch_ids`)?
The view:* fields can be provided in Item properties.
How useful is it to know the relative burst and polarization per beam?

**Comment by davidraleigh** [comment]
Is this useful for satellites other than sentinel-1?

**Comment by davidraleigh** [comment]
seems like this is maybe more appropriate for the sentinel-1 extension. unless there are other data providers where this is useful?

**Comment by emmanuelmathot** [comment]
The idea is to have an scalable description of the swath. Each beam is a specific steering of the instrument and thus has it's own parameters (i.e. view).
Concerning the search, I do not see any limitation or complexity for filtering

```A_EQUALS(sar:swath.*.id,["S1","S2"])```

Shall work like

```A_EQUALS(sar:subswath_ids,["S1","S2"])```

no?

**Comment by fabricebrito** [comment]
@davidraleigh It's relevant for all SAR missions. 

See pages 14-15 of https://www.asi.it/wp-content/uploads/2019/08/COSMO-SkyMed-Mission-and-Products-Description_rev3-2.pdf for CSK and this is just an example.

**Comment by m-mohr** [comment]
> Concerning the search, I do not see any limitation or complexity for filtering

@emmanuelmathot I'm not so sure. In theory yes, but in practice I still see implementations struggle to search bands or assets, so I don't think in practice you can currently easily search for the swath IDs in STAC API Filter implementations. Do you have different information? :-)

> beams and swath ids cannot be seen as synonyms

@fabricebrito Okay, I tried to understand this through some googling but can't find a good resource that explains the differences. I
I'm asking because I'm looking into CEOS ARD and they ask me to provide beam IDs, but I'm not sure how to provide that fos S1 for example, I thought it's swath IDs in the end. SAR is always so confusing ;-)

**Comment by fabricebrito** [comment]
> @fabricebrito Okay, I tried to understand this through some googling but can't find a good resource that explains the differences. I I'm asking because I'm looking into CEOS ARD and they ask me to provide beam IDs, but I'm not sure how to provide that for S1 for example, I thought it's swath IDs in the end. SAR is always so confusing ;-)
> 
> The other question is: If this is rather specific to one/some(?) SAR modes, should this go into a separate extension (e.g. ScanSAR)?

beam ids are relevant for StripMap acquisitions so I comes down to how the satellite is programmed to acquire data over the areas. Sentinel-1 Stripmap for instance is typically (not exclusively) used over small islands. 
My view is that at the sar extension level we need this information but this doesn't hamper having more advanced extensions for certain modes

**Review by fabricebrito — COMMENTED** [review]
We need a Stripmap example as this is the original request

File: examples/sentinel-1-grd-s4.json [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    15.424737
+  ],
+  "properties": {
+    "sat:absolute_orbit": 54112,
+    "sat:orbit_state": "descending",
+    "processing:datetime": "2024-05-31T11:25:04.541000+00:00",
+    "sar:instrument_mode": "SM",
+    "processing:level": "L1",
+    "sar:beam_ids": [
+      "S6"
```
Comment (fabricebrito) [inline examples/sentinel-1-grd-s4.json:52]: It's "S4"

**Review by fabricebrito — COMMENTED** [review]
@emmanuelmathot I hadn't seen the S1-S4-GRD example, it contains the wrong beam id (s6 instead of s4), so just a minor fix and we're good to go

**Comment by fabricebrito** [comment]
> I want to check whether this could be re-used for the CEOS ARD work, but this could take a bit longer... just fyi

will this create a delay in the approval?

**Review by fabricebrito — APPROVED** [review]
All good! Thanks @emmanuelmathot

**Pull request merged**